tesseract_python/tesseract_python/swig/tesseract_kinematics/core/forward_kinematics.i

```diff
@@ -0,0 +1,137 @@
+/**
+ * @file forward_kinematics.i
+ * @brief SWIG interface file for tesseract_kinematics/core/forward_kinematics.h
+ *
+ * @author John Wason
+ * @date December 10, 2019
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2019, Wason Technology, LLC
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+%{
+#include <tesseract_kinematics/core/forward_kinematics.h>
+%}
+
+%shared_ptr(tesseract_kinematics::ForwardKinematics)
+
+namespace tesseract_kinematics
+{
+class ForwardKinematics
+{
+public:
+  
+  using Ptr = std::shared_ptr<ForwardKinematics>;
+  using ConstPtr = std::shared_ptr<const ForwardKinematics>;
+
+  virtual ~ForwardKinematics() = default;
+
+%extend {
+
+  //virtual bool calcFwdKin(Eigen::Isometry3d& pose, const Eigen::Ref<const Eigen::VectorXd>& joint_angles) const = 0;
+  
+  Eigen::Isometry3d calcFwdKin(const Eigen::Ref<const Eigen::VectorXd>& joint_angles) const
```

Comment:
Why change the behavior?

Reply:
Python does not handle output parameters well. It would need to be converted to a tuple, or throw an exception on failure. The convention of Python is to throw an exception on error since exceptions are not very expensive in Python. I am probably going to change all the return codes to exceptions in the future, but I haven't had a chance to do so yet.

Reply:
Thanks for the explanation.